Automattic/jetpack
# Add URL navigation for the Jetpack Forms dashboard

Fixes #30314.

## Proposed changes:

This patch adds full support for URL based navigation for the new forms dashboard. All view settings are now persisted in the url, including: current tab & filters, current page and active response ID. This makes it much easier to access specific views/responses directly or share them with others.

This change also adds support for the browser's back button as each selection/filter/tab change now counts as a history item.

### Other information:


## Jetpack product discussion



pcsBup-1aJ-p2#comment-421

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






This is quite a large change so it's best to be thorough with testing:

- Go to `/wp-admin/admin.php?page=jetpack-forms`.
- You should notice the URL changing when you apply any filters, actively select a response, or change tabs.
- Refreshing the page should now bring you back to the exact same view you had before the refresh as all view settings are stored in the URL.
- Make sure date and source filters still work as expected.
- Make sure tabs still work as expected.
- Make sure we're always showing the correct active response.
- Make sure that the above settings are also correctly reflected when you export.
